models/Document/Page.php

```diff
@@ -274,4 +274,18 @@ public function getPreviewImageFilesystemPath()
     {
         return PIMCORE_SYSTEM_TEMP_DIRECTORY . '/document-page-previews/document-page-screenshot-' . $this->getId() . '@2x.jpg';
     }
+
+    public function save()
+    {
+        // Dispatch page preview message, if preview is enabled.
+        $documentsConfig = \Pimcore\Config::getSystemConfiguration('documents');
+        if ($documentsConfig['generate_preview'] ?? false) {
+
+            \Pimcore::getContainer()->get('messenger.bus.pimcore-core')->dispatch(
+                new GeneratePagePreviewMessage($this->getId(), \Pimcore\Tool::getHostUrl())
+            );
+        }
+
+        return parent::save();
```

Comment:
```suggestion
        parent::save();

        // Dispatch page preview message, if preview is enabled.
        $documentsConfig = \Pimcore\Config::getSystemConfiguration('documents');
        if ($documentsConfig['generate_preview'] ?? false) {
            \Pimcore::getContainer()->get('messenger.bus.pimcore-core')->dispatch(
                new GeneratePagePreviewMessage($this->getId(), \Pimcore\Tool::getHostUrl())
            );
        }

        return $this;
```

Reply:
What about doing this in an `DocumentEvents::POST_ADD`/`DocumentEvents::POST_UPDATE` event listener instead?

Reply:
@blankse is right. We should generate preview after calling parent::save() 👍. 

@jdreesen I'd not create a new listener for this task.